src/serde/read_cache_lookup.rs

```diff
@@ -162,12 +170,14 @@ impl ReadCacheLookup {
                     for (parent, direction) in items.iter() {
                         if *(self.count.get(parent).unwrap_or(&0)) > 0 && !seen_ids.contains(parent)
                         {
-                            if path.len() + 1 > max_path_length {
+                            if path.len() > max_path_length {
```

Comment:
There's an argument to be where with ties, we go with the back-ref, the idea being, it's no bigger but we end up with (maybe?) less memory used since we avoid allocating objects. Not really a deal-breaker though.

Reply:
yeah, I had the same thought. It probably makes sense to do that. I wanted to start with fixing the intention of the original code though.